client/asset/eth/eth.go

```diff
@@ -1395,18 +1395,19 @@ type gasEstimate struct {
 
 // initGasEstimate gets the best available gas estimate for n initiations. A
 // live estimate is checked against the server's configured values and our own
-// known values and errors or logs generated in certain cases.
+// known values and errors or logs generated in certain cases. It is an error if
+// the live gas estimate fails, so only use this method for order prep.
 func (w *assetWallet) initGasEstimate(n int, initVer, redeemVer, redeemAssetID uint32) (est *gasEstimate, err error) {
 	est = new(gasEstimate)
 
-	g := w.gases(initVer)
-	if g == nil {
+	// Get the refund gas.
+	if g := w.gases(initVer); g == nil {
 		return nil, fmt.Errorf("no gas table")
+	} else { // scoping g
+		est.Refund = g.Refund
 	}
 
-	est.Refund = g.Refund
-
-	est.Swap, est.nSwap, _, err = w.swapGas(n, initVer)
+	est.Swap, est.nSwap, _, err = w.swapGas(n, initVer, true) // fail if live gas estimate fails
```

Comment:
While you're here, a few lines down, it says
```Go
est.oneGas = est.Swap + est.Redeem
est.nGas = est.nSwap + est.nRedeem
```

Adding the redeem value there is wrong and pointless. They're both zero. I think those lines needs to come after the redeem wallet check.

Reply:
Good catch.  This would only affect self-trades with same asset or ETH+TOKEN, right?

Reply:
Like this? https://github.com/decred/dcrdex/pull/2118/commits/ba8b9fa1fa5f5e2e911fd023e1088fe030abfa92

I'm now wondering if `maxOrder` needs to include refund in it's `oneFee` value.

Reply:
> I'm now wondering if `maxOrder` needs to include refund in it's `oneFee` value.

Oh, @JoeGruffins is doing this in 7222c01dd512108a1e337b42d33a9b4e6617ce09.  Good.